### PR TITLE
Feature/implement django forms

### PR DESCRIPTION
**Pull request by sandstromviktor**
image

## Description
[JIRA TASK](https://scilifelab.atlassian.net/browse/SS-1002?atlOrigin=eyJpIjoiYzNjYmJjZDMzNGM0NDgwZWIzMjYzNDNkYzJiMDA5MjQiLCJwIjoiaiJ9)
I guess it is here now, the PR you've all been waiting to review. A mere 250 files changed, 120 commits and a complete redesign of half of serve... Sorry about that. 

## Checklist

_If you're unsure about any of the items below, don't hesitate to ask. We're here to help!
This is simply a reminder of what we are going to look for before merging your code._

- [X] This pull request is against **develop** branch (not applicable for hotfixes)
- [X] I have included a link to the issue on GitHub or JIRA (if any)
- [X] I have included migration files (if there are changes to the model classes)
- [X] I have included, reviewed and executed tests (unit and end2end) to complement my changes
- [X] I have updated the related documentation (if necessary)
- [X] I have added a reviewer for this pull request
- [X] I have added myself as an author for this pull request
- [X] In the case I have modified settings.py, then I have also updated the studio-settings-configmap.yaml file in serve-charts

### CODE REVIEW

**Comment by churnikov** [comment]
Jesus Christ

**Review by churnikov — APPROVED** [review]
I'll review some more, but here's my thoughts so far

File: apps/types_/app_registry.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+from apps.types_.app_types import ModelFormTuple, OptionalModelFormTuple
+
+
+class AppRegistry:
```
[the remark this answers is not in the document]
Comment (churnikov) [reply apps/types_/app_registry.py:4]: Add a method that groups models by category

File: apps/tasks.py [hunk 2]
```diff
@@ -1,532 +1,162 @@
...
-            logger.info("CALLING POST DELETE HOOKS")
-            post_delete_hooks(appinstance)
-        else:
-            status = AppStatus(appinstance=appinstance)
-            status.status_type = "FailedToDelete"
-            status.save()
-            appinstance.state = "FailedToDelete"
-        appinstance.save(update_fields=["state"])
+    # Handle deletion of apps in the "Develop" category
+    for subclass in BaseAppInstance.__subclasses__():
```
[the remark this answers is not in the document]
Comment (churnikov) [reply apps/tasks.py:34]: Method that groups could be used here

**Review by alfredeen — COMMENTED** [review]
Great work. This is a monster PR so somewhat difficult to review. But I have skimmed though the changes, have written some SQL against the new db schema and I am the author to some of the changes. All in all we have probably looked at this long enough and should now rely on the automated tests and then regression test this in staging.

File: apps/helpers.py [hunk 3]
```diff
@@ -7,69 +7,101 @@
...
+    "path": "Specify the path inside the container that you want to be persistent (path to database or similar). If \
+        you follow our guide to build the container, then please include the username in the path as well.",
+    "port": "Port that the docker container exposes and the application runs on. This should be an integer between \
+        3000-9999.",
+    "note_on_linkonly_privacy": "This option can be used only for a limited amount of time, for example while under \
+        development or during peer review.",
+}
+
+
+class SubdomainCandidateName:
```
[the remark this answers is not in the document]
Comment (alfredeen) [reply apps/helpers.py:55]: Just to be clear, by types, do you mean under /apps/types_/ ?

[the remark this answers is not in the document]
Comment (churnikov) [reply apps/helpers.py:55]: Yes:)

[the remark this answers is not in the document]
Comment (alfredeen) [reply apps/helpers.py:55]: I will do this

**Comment by sandstromviktor** [comment]
Jippie!

**Pull request merged**